dynamiqs/solvers/utils/batching.py

```diff
@@ -3,32 +3,46 @@
 import torch
 from torch import Tensor
 
-from ...utils.tensor_types import ArrayLike, to_tensor
-from .td_tensor import TDTensor
+from ...utils.tensor_types import ArrayLike, TDArrayLike, to_tensor
+from .td_tensor import TDTensor, to_td_tensor
 
 
-def batch_H(H: TDTensor) -> TDTensor:
+def batch_H(
+    H: TDArrayLike,
+    dtype: torch.dtype | None = None,
+    device: str | torch.device | None = None,
```

Comment:
Correct type is (because we already converted everyone in `options`)
```suggestion
    dtype: torch.complex64 | torch.complex128,
    device: torch.device,
```